File: 230112-dojo/java/app/src/main/java/Solution.java
class Solution {
    public int solution(int[][] sizes) {
        int width = 0, height = 0;

        for (int[] card : sizes) {
            width = Math.max(width, Math.max(card[0], card[1]));
            height = Math.max(height, Math.min(card[0], card[1]));
        }
        return width * height;
    }
}

File: 230112-dojo/javascript/solution.js
export default function solution(sizes) {
  const max = sizes.map((value) => Math.max(value[0], value[1]));
  const min = sizes.map((value) => Math.min(value[0], value[1]));

  const width = max.sort((a, b) => b - a)[0];
  const height = min.sort((a, b) => b - a)[0];

  return width * height;
}
